homeassistant/components/logbook.py

```diff
@@ -317,43 +379,70 @@ def humanify(hass, events):
                 }
 
 
+def _get_related_entity_ids(session, events_filter):
+    from homeassistant.components.recorder.models import States
+    from homeassistant.components.recorder.util import \
+        RETRIES, QUERY_RETRY_WAIT
+    from sqlalchemy.exc import SQLAlchemyError
+    import time
+
+    timer_start = time.perf_counter()
+
+    query = session.query(States).with_entities(States.domain,
```

Comment:
This seems kinda wild, you're quering all available entities? Why is that?

Reply:
i was looking for a fastest/easiest way to get a list of all related entity_id's. so i can add filtering by time to the query, but that would probably require join'ing events table, as the only reliable time markers are there (time_fired column)?
i think this should be quick even on large datasets (taking into account additional index). i have 1m+ states records and query ranges from 0.03 to 0.05 seconds. My assumption is that making simple indexed queries should be faster than doing joins and filtering by time.